### PR TITLE
feat(front-end): add WizardShuffleList component

### PR DESCRIPTION
**Pull request by edalex-ian**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included
- [x] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

Adds a new Wizard component: Shuffle List. Following this PR will be an integration PR.

Before a user adds values:

2021-11-02-161457_849x227_scrot

Once a user adds values (note they are sorted, and can be removed via the delete buttons):

2021-11-02-161516_830x366_scrot

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: react-front-end/tsrc/components/wizard/WizardShuffleList.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  const [newEntry, setNewEntry] = useState<string>("");
+
+  const handleOnChange = () =>
+    pipe(
+      newEntry,
+      O.fromPredicate((s) => !S.isEmpty(s)),
+      O.map((s) =>
+        pipe(values, RSET.insert(S.Eq)(s), (updatedValues) => {
+          onChange(updatedValues);
+          setNewEntry("");
```
Comment (PenghaiZhang) [inline react-front-end/tsrc/components/wizard/WizardShuffleList.tsx:74]: Do you think an `IO `makes more sense here ?

**Review by PenghaiZhang — APPROVED** [review]
LGTM!

File: react-front-end/tsrc/components/wizard/WizardShuffleList.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  mandatory,
+  values,
+  onChange,
+}: WizardShuffleListProps): JSX.Element => {
+  const [newEntry, setNewEntry] = useState<string>("");
+
+  const handleOnChange = () =>
+    pipe(
+      newEntry,
+      O.fromPredicate((s) => !S.isEmpty(s)),
```
[the remark this answers is not in the document]
Comment (edalex-ian) [reply react-front-end/tsrc/components/wizard/WizardShuffleList.tsx:70]: But I need it negated (`!`).

File: react-front-end/tsrc/components/wizard/WizardShuffleList.tsx (hunk 1 above)
Comment (edalex-ian) [reply react-front-end/tsrc/components/wizard/WizardShuffleList.tsx:74]: I did ponder that (for probably too much time), but then thought it didn't really fit as this is in a handler and the intention is to execute. So I'd have this pipe return an IO for no other reason then to have to immediately execute. As a result, I ended up with this.

But if I'm missing something, more than happy to hear.

**Comment by edalex-ian** [comment]
The failed run here all looks fine for the purposes of this change. (It just needs to timeout tweaks I did over on the ShuffleBox branch.)

**Pull request merged**